chain/bitcoind_client.go

```diff
@@ -684,20 +681,24 @@ mainLoop:
 
 			// Check if the block is logically next. If not, we
 			// have a reorg.
-			if block.Header.PrevBlock == bs.Hash {
+			c.bestBlockMtx.Lock()
+			bestBlock := c.bestBlock
+			if block.Header.PrevBlock == bestBlock.Hash {
 				// No reorg. Notify the subscriber of the block.
-				bs.Hash = block.BlockHash()
-				bs.Height++
-				bs.Timestamp = block.Header.Timestamp
-				_, err = c.filterBlock(block, bs.Height, true)
+				bestBlock.Hash = block.BlockHash()
+				bestBlock.Height++
```

Comment:
Fixed. A copy is made so that we can also use it within `c.reorg`. In the case that this is the next block, we'll filter the new block, modify the copy to be the new block, and finally assign it to `c.bestBlock`.